google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ReadWriteTransaction.java

```diff
@@ -644,11 +674,21 @@ public Void call() {
 
   @Override
   public ApiFuture<Void> commitAsync() {
-    checkValidTransaction();
+    checkValidTransaction(COMMIT_STATEMENT);
     state = UnitOfWorkState.COMMITTING;
     commitResponseFuture = SettableApiFuture.create();
     ApiFuture<Void> res;
-    if (retryAbortsInternally) {
+    // Check if this transaction actually needs to commit anything.
+    if (txContextFuture == null) {
```

Comment:
Just double checking my understanding: `checkValidTransaction(COMMIT_STATEMENT)` will initialize the `txContextFuture` here, so it won't be `null` for blind writes, correct?

Reply:
Correct, although it will only be initialized for blind writes if the transaction actually has any mutations. So a transaction that does not do anything except calling `Commit` will be a no-op (it already was).